src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -0,0 +1,49 @@
+package com.google.flourbot.execution;
+
+import com.google.flourbot.entity.EntityModuleImplementation;
+import com.google.flourbot.entity.Macro;
+import com.google.flourbot.entity.action.Action;
+import com.google.flourbot.entity.action.ActionType;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+// The Logic class of the server
+public class MacroExecutionModuleImplementation {
+
+  private final EntityModuleImplementation entityModuleImplementation;
+  // TODO: Add GoogleSheetHandler here when ready.
+
+  public MacroExecutionModuleImplementation(EntityModuleImplementation entityModuleImplementation)
+      throws Exception {
+    this.entityModuleImplementation = EntityModuleImplementation.getInstance();
+  }
+
+  public String execute(String userEmail, String message)
+      throws IllegalStateException, InterruptedException, ExecutionException {
+
+    String macroName = message.split(" ")[0];
+
+    Optional<Macro> optionalMacro = entityModuleImplementation.getMacro(userEmail, macroName);
+    if (!optionalMacro.isPresent()) {
+      throw new IllegalStateException("No macro named: " + macroName + " found!");
```

Comment:
no, if the macro is missing we should let the user know that they have misspelled the name, so we should return a message here.

Reply:
Done!